src/components/components-escolasCadastradas/TabelaEscola.tsx

```diff
@@ -0,0 +1,244 @@
+import "../../styles/App.css";
+import "../components-escolasCadastradas/TabelaEscola.css";
+import fetchlistSchools from "../../service/listSchools";
+import { useEffect, useState } from "react";
+import { EscolaData } from "../../models/service";
+import ExibirInformacoesEscola from "../../pages/ExibirInformacoesEscola";
+
+
+export default function TabelaEscola() {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+
+  const [showOptionsPages, setShowOptionsPages] = useState(false);
+  const [currentPage, setCurrentPage] = useState(1);
+  
+  const [showSchoolsPerPage, setShowSchoolsPerPage] = useState(false);
+  const [schoolsPerPage, setSchoolsPerPage] = useState(2);
+  const optionsSchoolsPerPage = ['2', '5', '10', '20'];
+  const [schools, setschools] = useState<EscolaData[]>([]);
+
+  const [modalStates, setModalStates] = useState(Array(schools.length).fill(false));
+
+  const handleButtonClick = (selectNumeber: any) => {
+    switch (selectNumeber) {
+      case 1:
+        setShowOptionsPages(!showOptionsPages);
+        break;
+      case 2:
```

Comment:
Não tem nenhuma função quando for case 2? acho que não tem necessidade de usar switch case se só tem função pro case 1, da pra ser feito só com um if